core/base-semantics-types.ts

```diff
@@ -2,4 +2,6 @@ export interface ISemanticsTyped {
 	$sType: string;
 }
 
-
+export interface IReference extends ISemanticsTyped {
```

Comment:
This type is now used in exactly one place. Do you know of an alternative way of expressing that type in that spot?